### PR TITLE
listener manager: removing exceptions

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low
Testing: updated tests
Docs Changes: n/a
Release Notes: n/a
https://github.com/envoyproxy/envoy-mobile/issues/176

### CODE REVIEW

File: source/common/listener_manager/listener_impl.cc [hunk 1]
```diff
@@ -236,8 +248,10 @@ std::string listenerStatsScope(const envoy::config::listener::v3::Listener& conf
     return absl::StrCat("envoy_internal_", config.name());
   }
   auto address_or_error = Network::Address::resolveProtoAddress(config.address());
-  THROW_IF_NOT_OK_REF(address_or_error.status());
-  return address_or_error.value()->asString();
+  if (address_or_error.status().ok()) {
+    return address_or_error.value()->asString();
+  }
+  return absl::StrCat("invalid_address_listener");
```
Comment (wbpcode) [inline source/common/listener_manager/listener_impl.cc:254]: Is this safe to don't throw or return an error here? And I think the absl::StrCat is unnecessary?

**Review by wbpcode — COMMENTED** [review]
LGTM overall with one comment. Thanks.

File: source/common/listener_manager/listener_impl.cc (hunk 1 above)
Comment (alyssawilk) [reply source/common/listener_manager/listener_impl.cc:254]: yeah this is fine - it basically defers the config load failure until slightly later when we use the invalid address.  arguably slightly better to fast fail but plumbing the error here requires a bunch of work so given the unliklihood of invalid IP address I figured the optimization wasn't worth the refactor.
That said while this is safe it's worth having a regression test, so I'll upload that shortly.

**Review by wbpcode — APPROVED** [review]

**Comment by wbpcode** [comment]
```
Code coverage for source/common/quic is lower than limit of 93.5 (93.4)
```

**Comment by alyssawilk** [comment]
@phlax how can I dig up coverage logs nowdaays? the link isn't working and I'm not seeing upload comments in the logs either

**Comment by alyssawilk** [comment]
Also this must be a flake - no code changes to quic, and I only added tests.
/retest

**Comment by phlax** [comment]
>  how can I dig up coverage logs nowdaays?

working on it right now - should be fixed today/tomorrow

**Comment by alyssawilk** [comment]
either way https://github.com/envoyproxy/envoy/pull/36398 should hopefully mitigate flake

**Comment by kyessenov** [comment]
/wait
coverage fix

**Review by wbpcode — APPROVED** [review]

**Pull request merged**